deob-bytecode/src/main/kotlin/org/openrs2/deob/bytecode/transform/StringDecryptionTransformer.kt

```diff
@@ -0,0 +1,183 @@
+package org.openrs2.deob.bytecode.transform
+
+import com.github.michaelbull.logging.InlineLogger
+import jakarta.inject.Singleton
+import org.objectweb.asm.Opcodes
+import org.objectweb.asm.tree.AbstractInsnNode
+import org.objectweb.asm.tree.FieldInsnNode
+import org.objectweb.asm.tree.FieldNode
+import org.objectweb.asm.tree.LdcInsnNode
+import org.objectweb.asm.tree.MethodInsnNode
+import org.objectweb.asm.tree.TableSwitchInsnNode
+import org.openrs2.asm.classpath.ClassPath
+import org.openrs2.asm.deleteExpression
+import org.openrs2.asm.intConstant
+import org.openrs2.asm.transform.Transformer
+
+@Singleton
+public class StringDecryptionTransformer : Transformer() {
+    private var stringsDecrypted = 0
+    private var classesDecrypted = 0
+    private var stringsInlined = 0
+
+    override fun preTransform(classPath: ClassPath) {
+        for (library in classPath.libraries) {
+            for (clazz in library) {
+                val clinit = clazz.methods.find { it.name == "<clinit>" } ?: continue
+
+                // step 1: identify the decryption methods
+                val inner = clinit.instructions.iterator().asSequence().filterIsInstance<MethodInsnNode>()
+                    .filter {
+                        it.owner == clazz.name && it.desc == "(Ljava/lang/String;)[C" &&
+                            it.next?.opcode == Opcodes.INVOKESTATIC
+                    }.firstOrNull()?.let {
+                        clazz.methods.find { method -> method.name == it.name && method.desc == it.desc }
+                    }
+                val outer = clinit.instructions.iterator().asSequence().filterIsInstance<MethodInsnNode>()
+                    .filter {
+                        it.owner == clazz.name && it.desc == "([C)Ljava/lang/String;" &&
+                            it.previous?.opcode == Opcodes.INVOKESTATIC
+                    }.firstOrNull()?.let {
+                        clazz.methods.find { method -> method.name == it.name && method.desc == it.desc }
+                    }
+
+                if (inner == null || outer == null) continue
+
+                // step 2: get the xor table from the outer method
+                val switch: TableSwitchInsnNode = outer.instructions.iterator().asSequence()
+                    .filterIsInstance<TableSwitchInsnNode>().firstOrNull() ?: continue
+
+                val xorLookup = ArrayList<Number>()
+                for (label in switch.labels) {
+                    xorLookup.add(label.next.intConstant ?: continue)
```

Comment:
could use `+=` instead of `add` here - might be a bit cleaner.

I'd also use `nextReal`, which will skip past one or more non-virtual instructions so it always gets the next actual instruction (e.g. if there are two labels adjacent to each other, or a label and a line number node, etc.)